### PR TITLE
Remove corresponding draft order lines when variant is removing

### PR DESCRIPTION
**Pull request by IKarbowiak**
When the variant is removing, the corresponding draft order lines too.

# Impact

* [ ] New migrations
* [ ] New/Updated API fields or mutations
* [ ] Deprecated API fields or mutations
* [ ] Removed API types, fields, or mutations
* [ ] Documentation needs to be updated

# Pull Request Checklist

* [ ] Privileged queries and mutations are guarded by proper permission checks
* [ ] Database queries are optimized and the number of queries is constant
* [ ] Database migration files are up to date
* [ ] The changes are tested
* [ ] GraphQL schema and type definitions are up to date
* [x] Changes are mentioned in the changelog

### CODE REVIEW

**Review by fowczarek — APPROVED** [review]

File: saleor/graphql/product/mutations/products.py [hunk 1]
```diff
@@ -1372,6 +1373,25 @@ def success_response(cls, instance):
...
+    @classmethod
+    def perform_mutation(cls, _root, info, **data):
+        node_id = data.get("id")
+        instance = cls.get_node_or_error(info, node_id, only_type=ProductVariant)
+
+        # get draft order lines for variant
+        line_pks = list(
+            order_models.OrderLine.objects.filter(
+                variant=instance, order__status=OrderStatus.DRAFT
+            ).values_list("pk", flat=True)
```
Comment (maarcingebala) [inline saleor/graphql/product/mutations/products.py:1385]: Minor thing but we could use the queryset that we have for drafts - `Order.objects.drafts()`.

**Review by maarcingebala — COMMENTED** [review]
I think we need this logic in a few more places:
- `productVariantBulkDelete` mutation
- deleting a product, its variant will be deleted as well
- on deleting product types, as it also removes products

If it's problematic for some reasons, we don't have to cover all cases in this one PR, but we will need to have a task for that.

File: saleor/graphql/product/mutations/products.py (hunk 1 above)
Comment (IKarbowiak) [reply saleor/graphql/product/mutations/products.py:1385]: But we need here `OrderLine` instances not `Order`.

Comment (maarcingebala) [reply saleor/graphql/product/mutations/products.py:1385]: Yep, sorry :)

**Review by maarcingebala — APPROVED** [review]

**Review by fowczarek — APPROVED** [review]

**Pull request merged**